superset-frontend/src/dashboard/components/SliceHeaderControls/index.tsx

```diff
@@ -573,15 +691,39 @@ const SliceHeaderControls = (props: SliceHeaderControlsPropsWithRouter) => {
         overlayStyle={dropdownOverlayStyle}
         trigger={['click']}
         placement="bottomRight"
+        visible={dropdownIsOpen}
       >
         <span
-          css={css`
+          css={() => css`
             display: flex;
             align-items: center;
           `}
           id={`slice_${slice.slice_id}-controls`}
           role="button"
           aria-label="More Options"
+          tabIndex={0}
+          onClick={() => {
+            toggleDropdown();
+          }}
+          onBlur={e => {
+            // close unless the dropdown menu is clicked
+            const relatedTarget = e.relatedTarget as HTMLElement;
+            if (
+              dropdownIsOpen &&
+              menuRef?.current?.props.id !== relatedTarget?.id
+            ) {
+              toggleDropdown({ close: true });
+            }
+          }}
+          onKeyDown={e =>
```

Comment:
I'm wondering if it wouldn't be better to bake all that logic in `NoAnimationDropdown`. We could add those new props (onKeyDown, onBlur) to `NoAnimationDropdown` child by calling `cloneElement`